sunpy/map/mapbase.py

```diff
@@ -222,31 +221,11 @@ def __init__(self, data, header, plot_settings=None, **kwargs):
         # TODO: This should be a function of the header, not of the map
         self._validate_meta()
 
-        if self.dtype == np.uint8:
-            norm = None
-        else:
-            # Put import here to reduce sunpy.map import time
-            from matplotlib import colors
-            norm = colors.Normalize()
-
-        # Visualization attributes
-        self.plot_settings = {'cmap': 'gray',
-                              'norm': norm,
-                              'interpolation': 'nearest',
-                              'origin': 'lower'
-                              }
-        if plot_settings:
-            self.plot_settings.update(plot_settings)
-
-        # Try and set the colormap. This is not always possible if this method
-        # is run before map sources fix some of their metadata, so
-        # just ignore any exceptions raised.
-        try:
-            cmap = self._get_cmap_name()
-            if cmap in sunpy_cm.cmlist:
-                self.plot_settings['cmap'] = cmap
-        except Exception:
-            pass
+        self.plotter = MatplotlibPlotter(self, plot_settings)
+
+    @property
+    def plot_settings(self):
+        return self.plotter.plot_settings
```

Comment:
This isn't in scope for this week or this PR but I swear we talked about removing this once.

Reply:
In Phase 4 (not in this PR), should a deprecation warning be added here to encourage people to use `self.plotter.plot_settings` rather than `self.plot_settings`?